addons/web/static/src/js/widgets/rainbow_man.js

```diff
@@ -0,0 +1,48 @@
+odoo.define('web.rainbow_man', function (require) {
+"use strict";
+
+var ajax = require("web.ajax");
+var Widget = require('web.Widget');
+var core = require('web.core');
+var qweb = core.qweb;
+var _t = core._t;
+
+ajax.loadXML("/web/static/src/xml/rainbow_man.xml", qweb);
+
+var RainbowMan = Widget.extend({
+    template: 'rainbow_man.notification',
+    /**
+     * @constructor
+     * @param {Object} [options] - key-value options to decide rainbowman behavior / appearance
+     */
+    init: function(options) {
+        var animation_speed = {slow: 3000, medium: 1500, fast:500, no: false };
+        this.options = _.defaults(_.pick(options, _.identity) || {}, {
+            fadeout: 'medium',
+            img_url: '/web/static/src/img/smile.svg',
+            message: _t('Well Done!'),
+        });
+        this.delay = animation_speed[this.options.fadeout];
+        this._super.apply(this, arguments);
+        // On hashchange, remove rainbowman if it is not removed in some case (when fadeout is set to 'no')
+        core.bus.on('clear_uncommitted_changes', this, this.destroy);
+    },
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function() {
+            self.$('.o_reward_msg_txt').append(self.options.message);
+            if (self.delay) {
+                setTimeout(function() {
+                    self.$el.addClass('o_reward_fading');
+                    setTimeout(function() {
+                        self.destroy();
+                    }, 4000); // destroy only after fadeout animation is completed
```

Comment:
self.delay is the time RainbowMan stays on screen before fading out, and 4000 is the approximate time of fading out animation ( after self.delay ) so here we make sure that widget is not destroyed before animation is completed.